### ISSUE TITLE
propArray / FormArray update on updation of RxFormGroup object using patchModelValue(...)

### ISSUE THREAD

**Issue by ShinRai1090** [rxweb/rxweb#306]
### Description
I have a RxFormGroup object (created from a model) with propArray / formArray within it. Whenever the user adds an item to the list, the array in the model is updated with a new item and patchModelValue(...) is used to update the form with the new changes (array insertion).

Currently the FormArray controls are not updated with the patched model object changes, i.e. if there were 2 items and a 3rd was added the FormArray controls length stays at 2 with the initial 2 items.

### Describe the solution you'd like
The FormArray controls within the RxFormGroup should reflect the changes in the value object, i.e. the new controls array should have 3 items in the list with appropriate form controls. And vice versa for removal of items from the array.

**Comment by ajayojha** [comment]
@ShinRai1090 , I was trying to integrate the feature in the lib but thought the above case can be achieved by pushing the new ```FormGroup``` in the ```FormArray```, it will automatically reflect the new instance in the existing array instance.

Here is the code.


```js
ngOnInit() {
        let user = new User();
         let hobby = new Hobby();
         user.hobbies = new Array<Hobby>();
         user.hobbies.push(hobby);
        this.userInfoFormGroup = <RxFormGroup>this.formBuilder.formGroup(user); 
    }
    

    addHobby(){
        let hobbyFormArray = this.userInfoFormGroup.controls.hobbies as FormArray
        hobbyFormArray.push(this.formBuilder.formGroup(Hobby))
    }
```

```html
<div>
  <form  [formGroup]="userInfoFormGroup">
    
            <button class="btn btn-primary" (click)="addHobby()" >Add</button>
            <br/>
            <br/>
      <div class="form-group" [formGroup]="hobbyGroup" *ngFor="let hobbyGroup of getFormArray()">
      <label>Hobby Name</label>
      <input type="text" formControlName="name" class="form-control"  />      
    </div>

    {{userInfoFormGroup.modelInstance | json}}

  </form>
</div>

```
whenever user click on the add button then it will add new ```FormGroup``` in the ```FormArray``` and add new instance in the existing array object.


Please refer this stackblitz link : https://stackblitz.com/edit/rxweb-formarray-push-example?file=src%2Fapp%2Fpatch-complete.component.ts

If still, the provided solution is not worth in your case then please suggest I will accommodate in the library :).

**Comment by ShinRai1090** [comment]
Hey @ajayojha,,

I have currently used that approach, but I would like if the library had a patchModel like behaviour where the form would reflect the changes in the model object passed. i.e. modification of values or addition / removal of items from array (formArray reference).

Currently I have a model (class object) which is pretty much self sufficient, i.e. it has the addition / deletion logic for the rows. So I tried using that and patching the form with the updated model, but the formArray does not reflect the changes in the array, even though it relfects the changes in the other formControls.

I think the reflection of changes in the model should be consistent throughout, so that users do not have to use an external approach to handle the formArray. The main aim of this package and the reason I love it is to reduce boiler plating, with this feature enhancement it'd do that for nested FormArray handling :)

**Comment by ajayojha** [comment]
@ShinRai1090 Thanks for sharing the input. I will enhance the ```patchModel``` method as per need :).

**Comment by ajayojha** [comment]
@ShinRai1090 The feature has been released in the 1.9.9 release. Please update the same. 
I haven't enhanced the code the of ```patchModelValue```, because patch it self represents for an update in the existing properties.

I have applied some different which is more intuitive than the ```patchModelValue```. Now you don't need to call any method of ```RxFormBuilder``` whenever you add a new instance in the array model it will automatically create the ```FormGroup``` and add into the ```FormArray```. Here is the code:

First, you have to mark as true the ```autoInstancePush`` at the global level. 
```js
ReactiveFormConfig.autoInstancePush = true
```
That's it. Push the instance into the model array it will automatically reflect in the FormArray. For better understanding see the below code:
#### Models
```js
export class Skill {

    @unique()
    skillName: string;
}

export class Employee {

    @propArray(Skill)
    skills: Skill[];

}
```

#### Component
```js
export class AppComponent implements OnInit {
    employeeFormGroup: FormGroup
    employee: Employee;
    constructor(
        private formBuilder: RxFormBuilder) { }

    ngOnInit() {
        ReactiveFormConfig.autoInstancePush = true
        this.employee = new Employee();
        this.employee.skills = new Array<Skill>();
        this.employeeFormGroup = this.formBuilder.formGroup(employee);
    }

    removeSkills(index) {
        this.employee.skills.splice(index,0)
    }

    addSkill() {
      this.employee.skills.push( new Skill());
    }

}
```

Please let me know if there is any issue. I am closing this ticket as no further action required from my side.

**Comment by gregoriusus** [comment]
But still, would be better, that it works also with patchValues. I am receiving object with one property as array from api and would like to update with patchValues, not with push...
